include/OpenSpaceToolkit/Astrodynamics/Trajectory/State.hpp

```diff
@@ -33,6 +33,8 @@ using ostk::physics::time::Instant;
 class State
 {
    public:
+    State(const Instant& anInstant, const VectorXd& aCoordinates, const Shared<const Frame>& aFrameSPtr);
```

Comment:
Sorry if I missed any conversation about this already 🙏, but is 1 frame necessarily sufficient to define a state? I'm thinking of a future scenario where we may have a position that's in like an ECI frame and a "pointing quaternion" in a body centered frame. 
Does a single frame allow us to track those coordinates together?

Reply:
If so, I think we should use two different States (in two different frames), otherwise we have a higher confusion risk (plus we'd be storing more frame pointers, if we were to associate a frame per "state property")

Reply:
It's a good point indeed. For now there is a single frame per State, this does not mean that there can be more in the future. But, at least for this PR, I would start with a single frame per State

Reply:
I honestly can't think if a scenario in which we'd need more than 2 frames anyway... I'm good with leaving this for later 👍

Reply:
Added to Open questions in the proposal -> https://loftorbital.atlassian.net/wiki/spaces/FD/pages/2181335613/FDEP-003+Extended+State#Open-Questions